include/swift/AST/Pattern.h

```diff
@@ -343,6 +360,10 @@ class TuplePattern final : public Pattern,
 };
 
 /// A pattern which binds a name to an arbitrary value of its type.
+///
+/// This represents the name of a single variable.
+/// e.g `var a: Int?` includes the NamedPattern `a`, as well as the TypedPattern
+/// representing `Int?`.
```

Comment:
This is a little bit ambiguous in that one might think that there are two disjoint patterns contained inside a `VarDecl`, but actually it is represented as a `TypedPattern "Int?" (NamedPattern "a")`.

I think it might be clearer to instead say something like:

```
/// A pattern carrying the name for a binding.
///
/// \code
/// var a: Int = 0
///   // a: Int → TypedPattern "Int" (NamedPattern "a")
/// if case let .some(b) = ... { ... }
///   // let .some(b) → BindingPattern (EnumElementPattern "some" (ParenPattern (NamedPattern "b")) 
/// \endcode
```